src/functional/ffront/decorator.py

```diff
@@ -94,6 +94,26 @@ def _filter_closure_vars_by_type(closure_vars: dict[str, Any], *types: type) ->
     return {name: value for name, value in closure_vars.items() if isinstance(value, types)}
 
 
+def _canonicalize_args(node_params, args, kwargs) -> tuple[tuple, dict]:
+    new_args = []
+
+    for param_i, param in enumerate(node_params):
+        if param.id in kwargs:
+            if param.id not in ["out", "domain"] and param_i < len(args):
+                raise ProgramTypeError(f"got multiple values for argument {param.id}.")
+            new_args.append(kwargs[param.id])
+            kwargs.pop(param.id)
```

Comment:
```suggestion
```
To avoid modifying state in the caller. Isn't necessary anyway right?

Reply:
if you don't pop the arguments, you the error in the comment below is triggered

Reply:
I see, instead of workings with `kwargs` use a copy, e.g.
```
new_kwargs = {**kwargs}
```